### PR TITLE
libbeat,packetbeat,x-pack/filebeat/input/netflow: unify gopacket dependencies

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This unifies and updates the gopacket dependencies for network traffic analysis.

## Why is it important?

Currently the project depends on three forks of the gopacket library. This complicates dependency management, and with many of the fixes now merged upstream it needlessly prevent us having access to new features of gopacket (though note the intention to make a canonical [elastic/gopacket](https://github.com/elastic/gopacket) fork and elastic/gopacket#3 — hopefully in the long term merging changes back upstream.)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

No specific checks.

## How to test this PR locally

standard testing.

## Related issues

- Relates elastic/integrations#2132
- Relates #29112
- Depends #29017
- Depends elastic/gopacket#3

## Use cases

N/A

## Screenshots

N/A

## Logs

N/A

### CODE REVIEW

File: packetbeat/sniffer/sniffer.go [hunk 1]
```diff
@@ -141,24 +140,22 @@ func New(
...
 	if s.config.Dumpfile != "" {
-		dumper, err = openDumper(s.config.Dumpfile, handle.LinkType())
+		f, err := os.Create(s.config.Dumpfile)
 		if err != nil {
 			return err
 		}
+		defer f.Close()
 
-		defer dumper.Close()
+		w = pcapgo.NewWriter(f)
```
Comment (efd6) [inline packetbeat/sniffer/sniffer.go:157]: `pcapgo.NewWriterNanos(f)` here?

**Comment by efd6** [comment]
Not to be merged before elastic/gopacket#3 and with a replace directive added to build against that.

**Comment by efd6** [comment]
E2E failure looks unrelated.

File: packetbeat/sniffer/sniffer.go (hunk 1 above)
Comment (andrewkroh) [reply packetbeat/sniffer/sniffer.go:157]: That sounds like a good idea.

**Review by andrewkroh — APPROVED** [review]
LGTM

**Pull request merged**